### PR TITLE
💄✨ Mobile/API | Networking V3

### PR DESCRIPTION
**Pull request by tkapa**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

#561 

> 2. What was changed?

Adds the `Networking` page to the application
Adds a `Network` Controller with a Query to get a required list of `NetworkProfileDto`
Removes the existing `SSW People` page

> 3. Did you do pair or mob programming?

✏️ No

image
_**Figure: Friends Page**_

image
_**Figure: To Meet Page**_

### CODE REVIEW

File: src/Application/Network/Queries/GetNetworkProfileList/GetNetworkProfileListQuery.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            .ToList();
+
+        var staffDtos = await _dbContext.StaffMembers
+            .Join(_dbContext.Users,
+                staff => staff.Email, 
+                user => user.Email,
+                (staff, user) =>
+                new
+                {
+                    UserId = user.Id,
```
Comment (matt-goldman) [inline src/Application/Network/Queries/GetNetworkProfileList/GetNetworkProfileListQuery.cs:44]: Why aren't you using an AutoMapper projection for this?

File: src/MobileUI/AppShell.xaml [hunk 2]
```diff
@@ -179,8 +179,8 @@
         </Tab>-->
 
 
-        <Tab Route="people" Icon="icon_people" Title="SSW People">
-            <ShellContent ContentTemplate="{DataTemplate pages:PeoplePage}" />
+        <Tab Route="people" Icon="icon_people" Title="Networking">
```
Comment (matt-goldman) [inline src/MobileUI/AppShell.xaml:182]: What is in the design for this, is it "Network" or "Networking"?

File: src/MobileUI/ViewModels/NetworkingPageViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    {
+        IsBusy = true;
+        if (Segments is null || Segments.Count() == 0)
+        {
+            Segments = new List<Segment>
+            {
+                new() { Name = "Friends", Value = NetworkingPageSegments.Friends },
+                new() { Name = "To Meet", Value = NetworkingPageSegments.ToMeet },
+                new() { Name = "SSW", Value = NetworkingPageSegments.SSW }
+                // new() { Name = "This Year", Value = NetworkingPageSegments.Other },
```
Comment (matt-goldman) [inline src/MobileUI/ViewModels/NetworkingPageViewModel.cs:48]: Should probably remove this comment

File: src/MobileUI/ViewModels/NetworkingPageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                break;
+            case NetworkingPageSegments.SSW:
+            case NetworkingPageSegments.Other:
+            default:
+                SearchResults = Profiles.ToObservableCollection();
+                break;
+        }
+    }
+    
+    // TODO: Implement Navigation to OthersProfilePage with NetworkingProfileDto
```
Comment (matt-goldman) [inline src/MobileUI/ViewModels/NetworkingPageViewModel.cs:94]: Is there a PBI to link to for this TODO?

**Review by matt-goldman — COMMENTED** [review]
This is awesome!

Left some comments for you to review.

File: src/MobileUI/ViewModels/NetworkingPageViewModel.cs (hunk 4 above)
Comment (tkapa) [reply src/MobileUI/ViewModels/NetworkingPageViewModel.cs:94]: Ah, this is actually done. Removed the comment.

File: src/MobileUI/ViewModels/NetworkingPageViewModel.cs (hunk 3 above)
Comment (tkapa) [reply src/MobileUI/ViewModels/NetworkingPageViewModel.cs]: Done

File: src/MobileUI/AppShell.xaml (hunk 2 above)
Comment (tkapa) [reply src/MobileUI/AppShell.xaml]: It is "Network" I've updated everything appropriately :)

File: src/Application/Network/Queries/GetNetworkProfileList/GetNetworkProfileListQuery.cs (hunk 1 above)
Comment (tkapa) [reply src/Application/Network/Queries/GetNetworkProfileList/GetNetworkProfileListQuery.cs]: Hi Matt,
I would not recommend using Automapper here, for example `user.Avatar ?? "v2sophie"`. This is logic that should not be in a mapping profile. Automapper makes it very difficult to debug, and hard to reason with the code. I don't think I could ever recommend using Automapper.

From Matt P

Comment (matt-goldman) [reply src/Application/Network/Queries/GetNetworkProfileList/GetNetworkProfileListQuery.cs]: Isn't that logic to display `v2sophie` if `Avatar` is null or empty handled in the UI? If not it should be.

Comment (tkapa) [reply src/Application/Network/Queries/GetNetworkProfileList/GetNetworkProfileListQuery.cs:45]: Hey Matt,

As per our conversation, we agree that the existing approach isn't the best, but might require Schema changes in order to work #759

**Review by matt-goldman — APPROVED** [review]
❤️ Nice work, LGTM! 🚢

**Pull request merged**